lib/rsvp/enumerator.js

```diff
@@ -91,11 +91,10 @@ Enumerator.prototype._settleMaybeThenable = function(entry, i) {
   if (resolve === originalResolve) {
     let then = getThen(entry);
 
-    if (then === originalThen &&
-        entry._state !== PENDING) {
+    if (then === originalThen && entry._state !== PENDING) {
       entry._onError = null;
       this._settledAt(entry._state, i, entry._result);
-    } else if (typeof then !== 'function') {
+    } else if (!isFunction(then)) {
```

Comment:
i would prefer to leave the inline form of this. That way this helper function doesn't consume any of the inlining budget.